volatility3/framework/layers/qemu.py

```diff
@@ -192,6 +235,63 @@ def _load_segments(self):
             else:
                 raise exceptions.LayerException(self._name, f'QEMU unknown section encountered: {section_byte}')
 
+        # If no architecture has been set, try to determine it using fallback mechanisms
+        if not self._architecture:
+            self._architecture = self._fallback_determine_architecture()
+            if self._architecture is None:
+                vollog.log(constants.LOGLEVEL_VV, f"QEVM architecture could not be determined")
+
+        # Once all segments have been read, determine the PCI hole if any
```

Comment:
When this code is executed before the segments are read, more images will work. I didn't figure out why though 🤔 However, moving this code up will make reading the architecture from the `QEVM_CONFIGURATION`-section useless, since the value is not needed anymore.

Reply:
That suggests that the value from the configuration and the value the fallback is finding don't match up, which is concerning, but at least an avenue for investigation.  I don't have time to dig into it now, but I'll give it a look when I can...

Reply:
The values do match. I think it is caused by the fact that the code for reading the segments makes some calls to `self._extract_data()`, which calls `self._get_ram_segments()`, which on its turn needs to know where the PCI-holes are. So reading segments before knowing the hole locations will lead to the images being read as if there are no holes?